### PR TITLE
Added Android ASAN repack script

### PR DESCRIPTION
**Pull request by JCash**
Added an easy-to-use .apk repack script for users that have built their custom engine with ASAN.
The script copies the correct shared library from a local install of the Android NDK in DYNAMO_HOME.

### CODE REVIEW

File: scripts/mobile/README_ANDROID.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+In game.project, set the property `android.debuggable` to `1`.
+
+Note: The wrap.sh support was added in Android O (API 27).
+
+[Documentation - wrap.sh](https://developer.android.com/ndk/guides/wrap-script#creating_the_wrap_shell_script)
+[Documentation - Asan](https://developer.android.com/ndk/guides/asan)
+
+UBSan is currently not supported by Android.
+
+### C++ and ASAN
```
Comment (JCash) [inline scripts/mobile/README_ANDROID.md:21]: The needed app manifest

File: scripts/mobile/README_ANDROID.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+## ASan
+
+The script `android-repack-asan.sh` doesn't actually replace the executable in the package.
+What it does it detect if the executable is dependent on the `libclang_rt.asan-*-android.so` library.
+If so, it will copy the file to the correct library path in the package.
+
+Also, it will then copy the `android-wrap.asan.sh` as the `wrap.sh` in the package.
+
+In order to run this package via the `wrap.sh`, you will need to have set the project to "debuggable".
+In game.project, set the property `android.debuggable` to `1`.
```
Comment (JCash) [inline scripts/mobile/README_ANDROID.md:12]: You need to set the project as debuggable.

File: scripts/mobile/android-repack-asan.sh [hunk 3]
```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env bash
```
Comment (JCash) [inline scripts/mobile/android-repack-asan.sh:1]: This script is mostly identical to `android-repack.sh`, but without copying the engine from DYNAMO_HOME.

If you wish to build a local engine with Asan, then run `android-repack.sh` first, then `android-repack-asan.sh` second.

File: scripts/mobile/android-repack.sh [hunk 4]
```diff
@@ -29,10 +29,10 @@ function terminate() {
...
 
 function terminate_usage() {
-    echo "Usage: ${SCRIPT_NAME} <source> [<certificate>, <key>]"
-    echo "  source       - absolute filepath to the source apk to repack"
-    echo "  certificate  - absolute filepath to the certificate file (pem)"
-    echo "  key          - absolute filepath to the keyfile (pk8)"
+    echo "Usage: ${SCRIPT_NAME} <source> <keystore> <keystore_pass>"
+    echo "  source        - absolute filepath to the source apk to repack"
+    echo "  keystore      - absolute filepath to the keystore"
+    echo "  keystore pass - absolute filepath to the keystore password"
```
Comment (JCash) [inline scripts/mobile/android-repack.sh:35]: Updated to use the new keystore instead.

File: scripts/mobile/android-wrap-asan.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#!/system/bin/sh
```
Comment (JCash) [inline scripts/mobile/android-wrap-asan.sh:1]: You can also have a `wrap.sh` that enables `malloc debug`:
* https://developer.android.com/ndk/guides/wrap-script#malloc_debug
* https://android.googlesource.com/platform/bionic/+/master/libc/malloc_debug/README.md

**Review by britzl — DISMISSED** [review]
Looks good!

**Review by AGulev — DISMISSED** [review]
👍

**Pull request merged**